src/MongoDB.Driver.Core/Core/Operations/ListCollectionsOperation.cs

```diff
@@ -122,11 +135,18 @@ private IReadOperation<IAsyncCursor<BsonDocument>> CreateOperation(IChannel chan
         {
             if (Feature.ListCollectionsCommand.IsSupported(channel.ConnectionDescription.ServerVersion))
             {
-                return new ListCollectionsUsingCommandOperation(_databaseNamespace, _messageEncoderSettings) { Filter = _filter };
+                return new ListCollectionsUsingCommandOperation(_databaseNamespace, _messageEncoderSettings)
+                {
+                    Filter = _filter,
+                    NameOnly = _nameOnly
+                };
             }
             else
             {
-                return new ListCollectionsUsingQueryOperation(_databaseNamespace, _messageEncoderSettings) { Filter = _filter };
+                return new ListCollectionsUsingQueryOperation(_databaseNamespace, _messageEncoderSettings)
+                {
+                    Filter = _filter
```

Comment:
We can't pass the NameOnly option here but it doesn't matter. We'll still just only use the "name" field of the response.